src/test/spec/v2_runner/mod.rs

```diff
@@ -140,6 +143,9 @@ pub async fn run_v2_test(test_file: TestFile) {
         let mut session1 = Some(client.start_session(options).await.unwrap());
         let session1_lsid = session1.as_ref().unwrap().id().clone();
 
+        // initialize with abortTransaction to account for implicit aborts in command monitoring
+        let mut command_names = vec!["abortTransaction".into()];
```

Comment:
It's because of [this test](https://github.com/mongodb/specifications/blob/33916af7c70397732c203b4c3f55c7bbf8e2aebc/source/transactions/tests/legacy/create-collection.json#L16) (and other similar tests); the `assertCollectionExists` and `assertCollectionNotExists` both use `list_collection_names` which emit command events that aren't included in the expected events. We can't filter these out indiscriminately because other tests using the v2 runner (e.g. retryable reads) do look for these in expected events.

Reply:
Ah I see. So I don't think the v2 runner mentions this explicitly, but the `testRunner` operations are supposed to use the "internal client" (or at least, _an_ "internal client" that's separate from the client used as part of the test execution). This is the same client used to create the collections at the beginning and to seed the initial data. The unified test format makes this requirement explicit: 

>The assertCollectionExists operation instructs the test runner to assert that the given collection exists in the database. The test runner MUST use the internal MongoClient for this operation.

Once this is done, we can be sure that the test runner assertions aren't interfering with any of the expected behavior or event assertions included in the actual test itself.